### PR TITLE
CI: Enable mypy

### LINKED ISSUE #180: Mypy support
This codebase currently has quite a few type annotations for classes and functions. Using a static type checker (mypy) to validate these would:

- prevent these type annotations from going out of date
- would allow us to pick up bugs in the software (or allow us to pick up edge cases, where we can provide better error messages)

### PR DESCRIPTION
**Pull request by VeckoTheGecko**
- Closes https://github.com/Parcels-code/virtualship/issues/180
- [x] Tests added - NA
- [x] Re-ran README.md help commands if the CLI has changed - NA
- [x] PR adheres with the current [design document](./design-doc.md) (or design document is updated)

This PR doesn't actually fix typing errors, just re-enables the workflows.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -69,51 +69,77 @@ filterwarnings = [
...
+  "src/virtualship/instruments/argo_float.py",
+  "src/virtualship/instruments/base.py",
+  "src/virtualship/instruments/ctd.py",
+  "src/virtualship/instruments/drifter.py",
+  "src/virtualship/instruments/ship_underwater_st.py",
+  "src/virtualship/instruments/xbt.py",
+  "src/virtualship/make_realistic/problems/simulator.py",
+  "src/virtualship/models/checkpoint.py",
+  "src/virtualship/models/expedition.py",
+  "src/virtualship/utils.py",
```
Comment (VeckoTheGecko) [inline pyproject.toml:148]: Gradually this list should become smaller as we become typesafe in virtualship

**Review by j-atkins — APPROVED** [review]
